### PR TITLE
Revise and improve the troubleshooting guide

### PR DESCRIPTION
**Pull request by zliu41**
Removed some outdated content, and added explanations for some common issues.

Closes https://github.com/IntersectMBO/plutus-private/issues/46

### CODE REVIEW

File: doc/docusaurus/docs/troubleshooting.md [hunk 1]
```diff
@@ -4,122 +4,54 @@ sidebar_position: 80
...
 
-Errors that start with `GHC Core to PLC plugin` are errors from `plutus-tx-plugin`.
+### "No unfolding"
 
-> :pushpin: **NOTE** 
-> 
-> Often, these errors arise due to GHC doing something to the code before the plugin gets to see it. 
-> The solution is often to prevent GHC from doing this. 
-> We often recommend trying various GHC compiler flags.
+This means the plugin cannot access to the definition of an identifier, which it needs to be able to compile the identifier to Plutus Core.
```
Comment (bezirg) [inline doc/docusaurus/docs/troubleshooting.md:11]:
```suggestion
This means the plugin cannot access to the definition of a Haskell identifier, which it needs to be able to compile that identifier to Plutus Core.
```

File: doc/docusaurus/docs/troubleshooting.md [hunk 2]
```diff
@@ -4,122 +4,54 @@ sidebar_position: 80
...
-To tell the plugin to defer any errors to runtime, add the following lines for your `package-name` to `cabal.project`:
+If this doesn't resolve the issue, or if the identifier in question isn't directly defined in the code but is produced by GHC optimizations,
+ensure that you apply all GHC flags listed in [Compiling Plutus Tx](./using-plutus-tx/compiling-plutus-tx.md).
+These flags disable GHC optimizations that can interfere with the plugin, and ensure that unfoldings are neither omitted nor ignored.
 
-```
-package package-name
-  haddock-options: "--optghc=-fplugin-opt PlutusTx.Plugin:defer-errors"
-```
+If the identifier with missing unfolding is from `base` or invoked by a function from `base`, you should use the corresponding function from the `plutus-tx` package.
```
Comment (bezirg) [inline doc/docusaurus/docs/troubleshooting.md:21]:
```suggestion
If the identifier with missing unfolding is from `base` or invoked by a function from `base`, you should use instead the corresponding function from the `plutus-tx` package.
```

File: doc/docusaurus/docs/troubleshooting.md [hunk 3]
```diff
@@ -4,122 +4,54 @@ sidebar_position: 80
...
+The plugin does not support many functions and methods from `base`.
```
Comment (bezirg) [inline doc/docusaurus/docs/troubleshooting.md:22]:
```suggestion
Note that the plugin does not support all functions and methods from `base`.
```

File: doc/docusaurus/docs/troubleshooting.md [hunk 4]
```diff
@@ -4,122 +4,54 @@ sidebar_position: 80
...
 
-### Non-`INLINABLE` functions
+This error can also occur if the identifier simply doesn't have an unfolding, e.g., `f x = $$(compile [|| x + 1 ||])`.
+Clearly there is no unfolding for `x`, so it is impossible for it to work.
 
-A common error is:
+Last but not the least, this error may happen when using GHCi, which is not fully supported by the plugin.
```
Comment (bezirg) [inline doc/docusaurus/docs/troubleshooting.md:27]:
```suggestion
Alternatively, this error may happen when using GHCi, which is not fully supported by the plugin.
```

**Review by bezirg — APPROVED** [review]

**Review by effectfully — APPROVED** [review]
Great stuff.

File: doc/docusaurus/docs/troubleshooting.md [hunk 5]
```diff
@@ -4,122 +4,54 @@ sidebar_position: 80
...
 
-For more advice on using Haskell Language Server (HLS), consult the [CONTRIBUTING guide](https://github.com/IntersectMBO/plutus/blob/master/CONTRIBUTING.adoc) in the `plutus` repository.
+### Missing Trace Messages
 
-### Wrong version
+If your expected trace messages are missing, check the following [plugin flags](./delve-deeper/plutus-tx-compiler-options.md):
 
-`ghcide compiled against GHC 8.10.3 but currently using 8.10.2.20201118`
+- If the `remove-trace` flag (default off) is on, all trace messages will be removed.
+- If the `preserve-logging` flag (default off) is off, the compiler may remove some trace messages during optimization.
```
[the remark this answers is not in the document]
Comment (zliu41) [reply doc/docusaurus/docs/troubleshooting.md:46]: Yeah the flags need some rework.

`remove-trace` removes all traces, while `preserve-logging` doesn't. The latter only removes trace if the plugin evaluates a builtin at compile time that produces traces. E.g., it evaluates `trace "hello" x`, but not `trace (show x) x`.

**Pull request merged**